db/silkworm/etl/collector.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+   Copyright 2020 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include "collector.hpp"
+
+#include <queue>
+#include <boost/filesystem.hpp>
+
+namespace silkworm::etl {
+
+    namespace fs = boost::filesystem;
+
+    Collector::~Collector()
+    {
+        if (work_path_.empty()) {
+            return;
+        }
+
+        fs::path path(work_path_);
+        if (fs::exists(path)) {
+            fs::remove_all(path);
+        }
+    }
+
+
+    void Collector::flush_buffer() {
+    if (buffer_.size()) {
+        buffer_.sort();
+        file_providers_.emplace_back(work_path_, (int)file_providers_.size());
```

Comment:
I'd change the type of FileProvider's id to `size_t` instead of `int` to avoid this type cast.